lib/hermit-git

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+#
+# Summary: Run git operations on current profile repository
+# Usage: hermit git <git arguments>
+
+# This is a pass-through from hermit to git. It doesn't implement
+# any additional functionality, just changes to the profile directory
+# and runs git with the arguments specified.
+#
+# Any commands that implement additional functionality should be broken
+# out of this file and implemented as their own wrappers to what git
+# provides.
+#
+# Copyright 2015, Aaron Griffin
+
+# This file is part of Hermit.
+
+# Hermit is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# Hermit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with Hermit. If not, see <http://www.gnu.org/licenses/>.
+
+set -e
+[ -n "$HERMIT_DEBUG" ] && set -x
+
+# Provide hermit completions
+if [ "$1" = "--complete" ]; then
+    profile_dir="$HERMIT_ROOT/profiles"
+    find "$profile_dir" -maxdepth 1 -type d | cut -c$((${#profile_dir}+2))-
+
+    exit
+fi
+
+check-git-config
+check-hermit-profile
+
+source $(dirname $0)/utilities
```

Comment:
One main reason to have tests for new commands is for sanity checks.  For instance, I just made some changes so that source `utilities` must come before calls to `check-git-config` and `check-hermit-profile` because the content of those files are now shell functions.